### DISCUSSION TITLE
GPU memory usage

### DISCUSSION THREAD

**Discussion by dang-trinh156** [lululxvi/deepxde#1413] [Q&A] [answered]
Dear Prof. Lu Lu, 
When I tried to run the example Kovasznay flow (https://github.com/lululxvi/deepxde/blob/master/examples/pinn_forward/Kovasznay_flow.py), the GPU memory usage is just around 2.5/24GB
![image](https://github.com/lululxvi/deepxde/assets/124559501/5a19d197-bd5e-485e-853e-47167b0c1012)
while Tensorflow 2.x allocates all possible GPU memory when running Automatic Differentiation (loss function, Neural Network backward) or simply just calculates derivative y_x of y = x**2. In other words, Tensorflow will use ~ 24GB/24GB no matter how small or simple your equation is.

Therefore, the question raised here is why DeepXDE runs on Tensorflow 2.x backend only uses 2.5/24GB while the Tensorflow library allocates all possible memory. Can you please kindly explain this difference between the two libraries?

Thank you for your answer!

**Comment by lululxvi** [comment]
See https://deepxde.readthedocs.io/en/latest/user/installation.html#tensorflow-1-x-backend

**Reply by dang-trinh156** [reply]
Thank you! But does 'TF_FORCE_GPU_ALLOW_GROWTH = True' allow tensorflow works the same way as Pytorch in allocating the GPU memory?

**Reply by lululxvi** [reply]
"the same way as Pytorch"? What is the "way" of Pytorch?

**Reply by dang-trinh156** [reply]
I mean, if we set 'TF_FORCE_GPU_ALLOW_GROWTH = True', Tensorflow backend allocates memory the same way as Pytorch allocates memory in the GPU ?

**Reply by lululxvi** [reply, marked as answer]
Yes.

**Reply by dang-trinh156** [reply]
Thank you!
